sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/SortBenchmark.scala

```diff
@@ -36,7 +36,8 @@ import org.apache.spark.util.random.XORShiftRandom
 class SortBenchmark extends BenchmarkBase {
 
   private def referenceKeyPrefixSort(buf: LongArray, lo: Int, hi: Int, refCmp: PrefixComparator) {
-    val sortBuffer = new LongArray(MemoryBlock.fromLongArray(new Array[Long](buf.size().toInt)))
+    val sortBuffer =
+      new LongArray(OnHeapMemoryBlock.fromArray(new Array[Long](buf.size().toInt)))
```

Comment:
shall we create a new constructor of `OnHeapMemoryBlock` that accepts a single size parameter?